From e96bdfbedba29a3ce8c87736bb36590c6f01ad3b Mon Sep 17 00:00:00 2001
From: renchenglei <chenglei.ren@intel.com>
Date: Tue, 15 Feb 2022 10:31:21 +0800
Subject: [PATCH] Enable rotation on external display

This is to help enable window rotation on external
display.

Tracked-On: OAM-101226
Signed-off-by: Wan Shuang <shuang.wan@intel.com>
---
 .../java/com/android/server/display/LocalDisplayAdapter.java     | 1 +
 1 file changed, 1 insertion(+)

diff --git a/services/core/java/com/android/server/display/LocalDisplayAdapter.java b/services/core/java/com/android/server/display/LocalDisplayAdapter.java
index 2c08420af42d..4daea004a79c 100644
--- a/services/core/java/com/android/server/display/LocalDisplayAdapter.java
+++ b/services/core/java/com/android/server/display/LocalDisplayAdapter.java
@@ -574,6 +574,7 @@ final class LocalDisplayAdapter extends DisplayAdapter {
                     mInfo.type = Display.TYPE_EXTERNAL;
                     mInfo.touch = DisplayDeviceInfo.TOUCH_EXTERNAL;
                     mInfo.flags |= DisplayDeviceInfo.FLAG_PRESENTATION;
+                    mInfo.flags |= DisplayDeviceInfo.FLAG_ROTATES_WITH_CONTENT;
                     mInfo.name = getContext().getResources().getString(
                             com.android.internal.R.string.display_manager_hdmi_display_name);
                 }
-- 
2.17.1

